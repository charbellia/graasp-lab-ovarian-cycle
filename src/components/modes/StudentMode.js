import React, { Component, Fragment } from 'react';
import MainView from '../views/MainView';
import Header from '../layout/Header';

// using class because in the future we will fetch from api
// eslint-disable-next-line react/prefer-stateless-function
export class StudentMode extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <MainView />
      </Fragment>
    );
  }
}

export default StudentMode;
