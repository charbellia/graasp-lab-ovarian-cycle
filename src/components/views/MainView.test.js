import React from 'react';
import { shallow } from 'enzyme';
import StudentMode from '../modes/StudentMode';

describe('<StudentMode />', () => {
  const component = shallow(<StudentMode />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('shows one <StudentMode /> component', () => {
    expect(component.find('div.App').length).toEqual(1);
  });
});
