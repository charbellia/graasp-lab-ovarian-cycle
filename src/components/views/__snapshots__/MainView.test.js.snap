// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainView /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainView
    classes={Object {}}
    dispatchOvulationState={[MockFunction]}
    dispatchPostOvulationState={[MockFunction]}
    dispatchPreOvulationState={[MockFunction]}
    svg={
      Selection {
        "_groups": Array [
          Array [
            null,
          ],
        ],
        "_parents": Array [
          <html>
            <head />
            <body />
          </html>,
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Wrapper
          dayCount={1}
          secreteLhFsh={false}
          secreteOestro={false}
          secreteProgest={false}
        />,
        <Wrapper
          dayCount={1}
          handleFullCycle={[Function]}
          handleOvulation={[Function]}
          handlePostOvulation={[Function]}
          handlePreOvulation={[Function]}
          handleStop={[Function]}
          isStarted={false}
          obserViewActive={true}
          ovulation={false}
          ovulationActive={false}
          postOvulation={false}
          postOvulationActive={false}
          preOvulation={false}
          preOvulationActive={false}
          secreteLhFsh={false}
          secreteOestro={false}
          secreteProgest={false}
        />,
        <WithStyles(Connect(Refresher)) />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dayCount": 1,
          "secreteLhFsh": false,
          "secreteOestro": false,
          "secreteProgest": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dayCount": 1,
          "handleFullCycle": [Function],
          "handleOvulation": [Function],
          "handlePostOvulation": [Function],
          "handlePreOvulation": [Function],
          "handleStop": [Function],
          "isStarted": false,
          "obserViewActive": true,
          "ovulation": false,
          "ovulationActive": false,
          "postOvulation": false,
          "postOvulationActive": false,
          "preOvulation": false,
          "preOvulationActive": false,
          "secreteLhFsh": false,
          "secreteOestro": false,
          "secreteProgest": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Wrapper
            dayCount={1}
            secreteLhFsh={false}
            secreteOestro={false}
            secreteProgest={false}
          />,
          <Wrapper
            dayCount={1}
            handleFullCycle={[Function]}
            handleOvulation={[Function]}
            handlePostOvulation={[Function]}
            handlePreOvulation={[Function]}
            handleStop={[Function]}
            isStarted={false}
            obserViewActive={true}
            ovulation={false}
            ovulationActive={false}
            postOvulation={false}
            postOvulationActive={false}
            preOvulation={false}
            preOvulationActive={false}
            secreteLhFsh={false}
            secreteOestro={false}
            secreteProgest={false}
          />,
          <WithStyles(Connect(Refresher)) />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dayCount": 1,
            "secreteLhFsh": false,
            "secreteOestro": false,
            "secreteProgest": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dayCount": 1,
            "handleFullCycle": [Function],
            "handleOvulation": [Function],
            "handlePostOvulation": [Function],
            "handlePreOvulation": [Function],
            "handleStop": [Function],
            "isStarted": false,
            "obserViewActive": true,
            "ovulation": false,
            "ovulationActive": false,
            "postOvulation": false,
            "postOvulationActive": false,
            "preOvulation": false,
            "preOvulationActive": false,
            "secreteLhFsh": false,
            "secreteOestro": false,
            "secreteProgest": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
